assets/src/models/searchQuery.ts

```diff
@@ -9,15 +9,23 @@ export const searchPropertyDisplayConfig = {
 
 export type SearchProperty = keyof typeof searchPropertyDisplayConfig
 
+export type VehiclePropertyQuery = "all" | "vehicle" | "run" | "operator"
+export type SearchPropertyQuery = "all" | VehiclePropertyQuery | "location"
```

Comment:
I struggled with naming here. I think we'll want 3 different types: 
* individual properties excluding all (currently `SearchProperty`)
* all possible vehicle field queries including all (`VehiclePropertyQuery`)
* all possible vehicle + location field queries (`SearchPropertyQuery`)

Maybe renaming `SearchProperty` something like `SearchFields` would convey that those are the individual fields that can be queried, and the other two can become `VehicleProperty` and `SearchProperty`? That doesn't quite feel right to me either - any alternatives come to mind?

Reply:
I don't have any objections, so far I don't have any ideas for how it should be named.